src/daft-connect/src/session.rs

```diff
@@ -8,6 +8,12 @@ pub struct Session {
     /// Also, <https://users.rust-lang.org/t/hashmap-vs-btreemap/13804/4>
     config_values: BTreeMap<String, String>,
 
+    #[expect(
+        unused,
+        reason = "this will be used in the future especially to pass spark connect tests"
+    )]
+    tables_by_name: HashMap<String, daft_table::Table>,
+
```

Comment:
couldn't we just track the logical plan instead of the `Table` and defer execution? Storing all of the materialized tables in a hashmap will have a **huge** memory footprint.

Reply:
this is actually not relevant to this PR but should be discussed in the future  5aa8f7ab8041291c807bd5717378b2b953cee676